### DISCUSSION TITLE
Path cannot be empty

### DISCUSSION THREAD

**Discussion by alf07** [dompdf/dompdf#3338] [Q&A] [answered]
after connecting cloudflare, when trying to generate and download a pdf, I get an error

```
[ValueError] 
Path cannot be empty (0)
/var/www/www-root/data/www/****/local/lib/dompdf/vendor/phenx/php-font-lib/src/FontLib/BinaryStream.php:69
#0: fopen(string, string)
	/var/www/www-root/data/www/****/local/lib/dompdf/vendor/phenx/php-font-lib/src/FontLib/BinaryStream.php:69
#1: FontLib\BinaryStream->open(boolean, string)
	/var/www/www-root/data/www/****/local/lib/dompdf/vendor/dompdf/dompdf/lib/Cpdf.php:1187
#2: Dompdf\Cpdf->processFont(integer, array)
	/var/www/www-root/data/www/****/local/lib/dompdf/vendor/dompdf/dompdf/lib/Cpdf.php:942
#3: Dompdf\Cpdf->o_font(integer, string)
	/var/www/www-root/data/www/****/local/lib/dompdf/vendor/dompdf/dompdf/lib/Cpdf.php:3206
#4: Dompdf\Cpdf->output(boolean)
	/var/www/www-root/data/www/****/local/lib/dompdf/vendor/dompdf/dompdf/src/Adapter/CPDF.php:912
#5: Dompdf\Adapter\CPDF->stream(string, array)
	/var/www/www-root/data/www/****/local/lib/dompdf/vendor/dompdf/dompdf/src/Dompdf.php:869
#6: Dompdf\Dompdf->stream(string)
	/var/www/www-root/data/www/****/products/download_price.php:174
```

on the site there is an exact copy but without using cloudflare everything works fine.
Can anyone tell me what the problem is?

**Comment by bsweeney** [comment]
What do you mean by "an exact copy?" Just speculation because I haven't confirmed this would cause the problem, but it looks like maybe Dompdf is trying to use a font (i.e., the font is listed in installed-fonts.json) where the font file itself is not present? Like maybe if Dompdf was copied in full from one system to another and the font path is no longer valid?

**Reply by alf07** [reply, marked as answer]
the problem was in the Apache configs

there must be
`php_admin_value open_basedir "none"`

**Reply by bsweeney** [reply]
Something I pretty much always forget to ask someone to check. Thanks for the follow up.
